iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java

```diff
@@ -172,6 +175,15 @@ public EnvTree ENV() {
     );
   }
 
+  public UserTree USER() {
+    return b.<UserTree>nonterminal(DockerLexicalGrammar.USER).is(
+      b.firstOf(
+        f.user(b.token(DockerKeyword.USER), b.token(STRING_LITERAL_NO_COLON), b.token(Punctuator.COLON), b.token(STRING_UNTIL_EOL)),
+        f.user(b.token(DockerKeyword.USER), b.token(STRING_UNTIL_EOL))
```

Comment:
Maybe you have a look at the [grammar](https://github.com/camdencheek/tree-sitter-dockerfile/blob/main/src/grammar.json#L477) to avoid haveing yet another string literal pattern. And so you could use Optionals.